modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java

```diff
@@ -872,23 +876,234 @@ public void testBindPublishedObjectDefinitions() throws Exception {
 
 		// Bind two published object definitions
 
+		ObjectDefinition objectDefinitionB =
+			_addAndPublishCustomObjectDefinition("B");
+		ObjectDefinition objectDefinitionBB =
+			_addAndPublishCustomObjectDefinition("BB");
+
 		_testBindObjectDefinitions(
-			_addAndPublishCustomObjectDefinition("A"),
-			_addAndPublishCustomObjectDefinition("AA"),
+			objectDefinitionB, objectDefinitionBB,
 			(objectDefinition1, objectDefinition2) ->
 				TreeTestUtil.assertObjectDefinitionTree(
 					LinkedHashMapBuilder.put(
-						"A", new String[] {"AA"}
+						"B", new String[] {"BB"}
 					).put(
-						"AA", new String[0]
+						"BB", new String[0]
 					).build(),
 					_treeFactory.createObjectDefinitionTree(
 						objectDefinition1.getObjectDefinitionId(),
 						_objectDefinitionLocalService::getObjectDefinition),
 					_objectDefinitionLocalService));
 
+		// The object definitions must have the same scope
```

Comment:
// Unable to bind object definitions when they have different scopes"

I think it is best to follow the exception message because it explains better;

The actual message needs to clarify what we are trying to do so that the object definitions must have the same scope 
// The object definitions must have the same scope
> for what?